docs/infrastructure/microk8s-quickstart.md

```diff
@@ -0,0 +1,348 @@
+
+Assuming that you are starting from a blank slate, the first step is to install microk8s. Otherwise, please jump on to the [Configuring microk8s](###Configuring-microk8s) section.
+
+## 3. Installing microk8s
+
+1. Make sure the machine is up to date and also has the `nfs-common` package installed
+   >apt update && apt install nfs-common -y
+
+2. Install microk8s
+   >sudo snap install microk8s --classic --channel=1.15/stable
+   1. We recommend following the official documentation on how to [install microk8s](https://microk8s.io/docs/).
+
+## 4. Configuring microk8s
+
+1. First, we must enable the following plugins on microk8s:
+    >sudo microk8s.enable dns && sudo microk8s.status --wait-ready
+
+    >sudo microk8s.enable storage && sudo microk8s.status --wait-ready
+
+    >sudo microk8s.enable ingress &&  sudo microk8s.status --wait-ready
+
+2. Now that these plugins have been enabled, we should restart microk8s to ensure a proper bootstrap of the cluster
+    >microk8s.stop && microk8s.start && microk8s.status --wait-ready
+
+    Further information on this subject can be [found here](https://github.com/ubuntu/microk8s/issues/493#issuecomment-498167435).
+3. The plugins are now enabled and the cluster bootstrapped. However, we must still wait for some microk8s internals (dns, http, and ingress) plugins to be ready. Failing to do so can result in pods entering a `CrashLoopBackoff` state:
+
+    >microk8s.kubectl wait -n kube-system --for=condition=Ready pod -l k8s-app=kube-dns --timeout=300s
+
+    >microk8s.kubectl wait -n kube-system --for=condition=Ready pod -l k8s-app=hostpath-provisioner --timeout=300s
+
+    >microk8s.kubectl wait -n ingress --for=condition=Ready pod -l name=nginx-ingress-microk8s --timeout=300s
+
+4. Export your microk8s cluster configuration to the local kubeconfig file
+    >microk8s.config > ~/.kube/config
+
+After runing these commands, we have ensured that dns, http, and nginx ingress are enabled and working properly inside the cluster.
+
+## 5. Installing Codacy
+
+### 1. Helm and Tiller
+
+First we must install `helm` onto the cluster. With `helm`, we can easily install charts and manage the lifecycle of the installed artifacts. One example is the ability to rollback a failed install.
+
+1. Assuming you would want to install `helm` version 2.16.3, export the following environment variable
+    >export HELM_VERSION=2.16.3
+   1. Please note that we currently do *not* support helm v3.
+2. Run this script that will retrieve the desired `helm` version and install it onto the machine's `/usr/local/bin`:
+   1. Define a HELM_PKG variable
+      >HELM_PKG=helm-v${HELM_VERSION}-linux-amd64.tar.gz
+   2. Get the package
+      >wget https://get.helm.sh/$HELM_PKG
+   3. Extract the tarball
+      >tar xvzf $HELM_PKG
+   4. Move the binaries to `/usr/local/bin`
+      >mv linux-amd64/tiller linux-amd64/helm /usr/local/bin
+   5. Delete the extracted tarball directory
+      >rm -rvf $HELM_PKG linux-amd64/
```

Comment:
Include all the commands in a fenced code block so they're easier to copy+paste at once. We can still explain the main steps of the script below the code block.